testutil/testnode/query.go

```diff
@@ -0,0 +1,23 @@
+package testnode
+
+import (
+	"context"
+	"encoding/hex"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	rpctypes "github.com/tendermint/tendermint/rpc/core/types"
+)
+
+func QueryTx(clientCtx client.Context, hashHexStr string, prove bool) (*rpctypes.ResultTx, error) {
+	hash, err := hex.DecodeString(hashHexStr)
+	if err != nil {
+		return nil, err
+	}
+
+	node, err := clientCtx.GetNode()
+	if err != nil {
+		return nil, err
+	}
+
+	return node.Tx(context.Background(), hash, prove)
```

Comment:
Does `clientCtx` have an underlying go `Context`?

Reply:
hmm I thought so too, but myself and my IDE could not find it? maybe some other Contetx does and we should refactor to that?

https://github.com/celestiaorg/cosmos-sdk/blob/50f4c49034ee97424a43b32e77715f9c8413bb5d/client/context.go#L25-L63